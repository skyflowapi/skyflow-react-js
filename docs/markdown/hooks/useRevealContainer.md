---
id: useRevealContainer
title: useRevealContainer
sidebar_label: useRevealContainer
---

{% env enable="reactJsSdkRef" %}

# useRevealContainer

sample documentation for useRevealContainer hook

## Import

```js
import {useRevealContainer} from 'skyflow-react-js';
```

## Returns
RevealContainer

{% /env %}