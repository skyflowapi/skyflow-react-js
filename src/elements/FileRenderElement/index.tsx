/*
  Copyright (c) 2022 Skyflow, Inc. 
*/
import React, { FC } from 'react'
import { FileRenderElements, SkyflowRenderElementProps } from '..'
import { v4 as uuid } from 'uuid';
import useUpdateFileRenderElement from '../../hooks/UpdateFileRenderElement';

const FileRenderElement: FC<SkyflowRenderElementProps> = ({ ...props }) => {
  const uniqueDivId = React.useRef(uuid());
  const [element, setElement] = React.useState<any>(null);

  React.useEffect(() => {
    const divElement = document.getElementById(props?.id || `reveal-${uniqueDivId.current}`);
    try {
      if (divElement) {
        divElement.innerHTML = ''
      }
      const FileRenderElement = props?.container.create(
        {
          ...props.classes,
          altText: props.altText || '',
          skyflowID: props.skyflowID,
          column: props.column,
          table: props.table,
        }
      )
      FileRenderElement.mount(props.id ? `#${props.id}` : `#reveal-${uniqueDivId.current}`)
      FileRenderElements[props.id as string] = FileRenderElement
      setElement(FileRenderElement);
    } catch (e) {
      console.error(e)
    }
  }, [])

  useUpdateFileRenderElement(props, element)
  return <div id={props.id ? props.id : `reveal-${uniqueDivId.current}` }></div>
}

export default React.memo(FileRenderElement);
