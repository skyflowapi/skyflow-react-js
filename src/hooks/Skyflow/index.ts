import { useSkyflowClient } from '../../core/hook';
import React from 'react';
import Skyflow from 'skyflow-js';

const useSkyflow = (): Skyflow => {
    const { skyflow } = useSkyflowClient()
    
    return React.useMemo(
    () => skyflow as Skyflow,
    [],
    )
}

export default useSkyflow
