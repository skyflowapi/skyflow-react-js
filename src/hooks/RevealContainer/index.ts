/*
  Copyright (c) 2022 Skyflow, Inc. 
*/
import React from 'react'
import Skyflow from 'skyflow-js'
import RevealContainer from 'skyflow-js/types/core/external/reveal/reveal-container'

import { useSkyflow } from '../../core/SkyflowElements/hook'

/**
 * Container for Reveal Elements.
 * @returns RevealContainer
 */
export const useRevealContainer = (): RevealContainer => {
  const { skyflow } = useSkyflow()
  return React.useMemo(
    () => skyflow?.container(Skyflow.ContainerType.REVEAL) as RevealContainer,
    [],
  )
}
