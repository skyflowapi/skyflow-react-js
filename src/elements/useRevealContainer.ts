import Skyflow from 'skyflow-js';
import RevealContainer from 'skyflow-js/types/core/external/reveal/RevealContainer';
import { useSkyflow } from '../core/SkyflowElements';

const useRevealContainer = (): RevealContainer => {
  const { skyflow } = useSkyflow();

  return skyflow.container(Skyflow.ContainerType.REVEAL) as RevealContainer;
};

export { useRevealContainer };
