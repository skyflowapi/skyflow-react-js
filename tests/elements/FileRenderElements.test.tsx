/*
	Copyright (c) 2022 Skyflow, Inc. 
*/
import * as React from 'react'
import { render } from '@testing-library/react'
import FileRenderElement from '../../src/elements/FileRenderElement'
import RevealContainer from 'skyflow-js/types/core/external/reveal/reveal-container'
import Skyflow from 'skyflow-js'
import useRevealContainer from '../../src/hooks/RevealContainer'

const mountMock = jest.fn();
const eventListenerMock = jest.fn();

const revealElementMock = {
  mount: mountMock,
  on: eventListenerMock
}

jest.mock('../../src/hooks/RevealContainer', () => ({
  __esModule: true,
  default: (() => ({
    create: () => (revealElementMock),
    type: 'REVEAL'
  }))
}))
jest.mock('uuid',()=>{return {
  v4 :()=>('12443')
}
})

describe('RevealElement', () => {
  let container: RevealContainer
  let id: string
  let classes: Record<string, unknown>
  let collectContainer;

  test('should reveal element', async () => {
    const revealContainer = useRevealContainer();

    const renderElement = render(
    <FileRenderElement container={revealContainer} id={'id'} skyflowID={'id'} column={'column'} table={'table'}/>,
    )
    expect(renderElement).toMatchSnapshot()
  })

  test('should test file render element without id', () => {
    const revealContainer = useRevealContainer();
    const renderElement = render(
    <FileRenderElement container={revealContainer} id={''} classes={classes} skyflowID={'id'} column={'column'} table={'table'}/>,
    )
    expect(renderElement).toMatchSnapshot()
  })

  test('should  test file render element error scenario', () => {
    const renderElement = render(
    <FileRenderElement container={collectContainer} id={'123'} classes={classes} skyflowID={'id'} column={'column'} table={'table'}/>,
    )
    expect(renderElement).toMatchSnapshot()
  })
})
