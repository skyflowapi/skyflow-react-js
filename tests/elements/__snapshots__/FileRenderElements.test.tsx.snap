// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevealElement should  test file render element error scenario 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="123"
      />
    </div>
  </body>,
  "container": <div>
    <div
      id="123"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RevealElement should reveal element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="id"
      />
    </div>
  </body>,
  "container": <div>
    <div
      id="id"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RevealElement should test file render element without id 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="reveal-12443"
      />
    </div>
  </body>,
  "container": <div>
    <div
      id="reveal-12443"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
